grabbit/core.py

```diff
@@ -185,9 +188,19 @@ def index(self):
         # Loop over all files
         for root, directories, filenames in os.walk(self.root, topdown=True):
 
-            # Exclude directories from further search if they match exclude regex
-            if self.exclude_dir is not None:
-                directories[:] = [d for d in directories if not re.match(self.exclude_dir, d)]
+            # Exclude/include top-level directories from further search
+            #if they match exclude regex
+            if root == self.root:
+                if self.index_regex:
+                    print(root)
+                    if 'exclude' in self.index_regex:
+                        for exclude_regex in self.index_regex['exclude']:
+                            directories[:] = [d for d in directories \
```

Comment:
Make sure to `break` once a match is found; no need to waste cycles looping over all possible candidates once we know the candidate is to be excluded.

Reply:
Once directories are excluded, the children are no longer looped over the way its written.